### ISSUE TITLE
Many GeoJSON markers with icons make page freeze

### ISSUE THREAD

**Issue by sgofferj** [Leaflet/Leaflet#7495]
- [X] I've looked at the [documentation](http://leafletjs.com/reference.html) to make sure the behavior is documented and expected
- [X] I'm sure this is a Leaflet code issue, not an issue with my own code nor with the framework I'm using (Cordova, Ionic, Angular, React…)
- [X] I've searched through the issues to make sure it's not yet reported

**Steps to reproduce**
- Added GeoJSON overlay with Finnish Roadworks
- Panned map around a bit
-> Freeze

**Expected behavior**
Leaflet shouldn't freeze
**Current behavior**
Leaflet freezes, browser opens popup complaining about unresponsive page

**Environment**
- Leaflet version: Leaflet 1.5.1+Detached: 2e3e0ffbe87f246eb76d86d2633ddd59b262830b.2e3e0ff
- Browser (with version): Chrome Version 89.0.4389.82 (Official Build) (64-bit)
- OS/Platform (with version): Ubuntu 20.04

**Additional context**
GeoJSON URL: https://tie.digitraffic.fi/api/v2/data/maintenance/trackings/latest
Data is CC-BY-4.0

**Minimal example reproducing the issue**
Well, I wrote the example and the plunker tab froze...

Here are the code snippets in question:

```
const roadworksIcon = L.divIcon({
    html: '<i class="fa fa-warning"></i>',
    iconSize: [20, 20],
    className: 'myicon roadworks'
});
```
Icon uses Font Awesome but that doesn't matter. It also freezes if I just use an asterisk.

```
var roadworks = new L.GeoJSON.AJAX("https://tie.digitraffic.fi/api/v2/data/maintenance/trackings/latest",{
  pointToLayer: function(geoJsonPoint, latlng) {
    return L.marker(latlng, { icon: roadworksIcon });
  },
  attribution: 'Roadworks data: Finnish Transport Infrastructure Agency&nbsp;CC-BY 4.0',
});
```
Similar issue. I'm using the AJAX function but the problem occurs without. It's just about the amount of markers in the file.

- [x] this example is as simple as possible
- [ ] this example does not rely on any third party code

**Comment by johnd0e** [comment]
Please check with latest lealfet

**Comment by sgofferj** [comment]
I upgraded Leaflet but at this moment, there is not much roadwork going on. Not as much as yesterday.
@johnd0e What is preferred? Leave this open for a moment until the roadworks increase (might take a couple of days) and I can test or close it for now and reopen if the issue reoccurs?

**Comment by johnd0e** [comment]
How many markers? If you mean thousands, then that freeze is expected, use https://leafletjs.com/reference-1.7.1.html#map-prefercanvas option to improve performance.

Feel free to reopen.

**Comment by sgofferj** [comment]
@johnd0e 
I'll try the prefercanvas option. At this moment there's 283 features in that roadworks GeoJSON and it makes 1.7.1 freeze when panning the map around a bit.. It doesn't freeze right away, though.

I downloaded a snapshot of the JSON: [latest.zip](https://github.com/Leaflet/Leaflet/files/6106898/latest.zip)

**Comment by sgofferj** [comment]
Ummm, no. With prefercanvas it freezes right away, with just one move...

![Screenshot_20210309_102241](https://user-images.githubusercontent.com/3258466/110440671-944c3d80-80c1-11eb-9cee-6ecf9c2641ee.png)

**Comment by sgofferj** [comment]
I worked around the issue for me now by putting the markers into an L.markerClusterGroup().
I'll still gladly help debugging as well as I can if you keep the issue open.

**Comment by johnd0e** [comment]
> I'll still gladly help debugging as well as I can if you keep the issue open.

Could you setup your problem sample code online?

**Comment by sgofferj** [comment]
Do you need just the relevant code or the actual running app?

**Comment by johnd0e** [comment]
I want you to setup very minimal (but running) code sample, that is enough to reproduce the issue.

**Comment by sgofferj** [comment]
Ok, first time I use Plunker, so I hope that works:
https://plnkr.co/edit/TWuCUFcZUNt5AuKd

**Comment by johnd0e** [comment]
> Ok, first time I use Plunker, so I hope that works:

Yes, that works, and does not freeze.
Perhaps that depends on some specific geojson content. so instead of querying fresh data you better stick to fixed set (like your  previous `latest.zip`)

**Comment by sgofferj** [comment]
I modified the Plunker project accordingly now. It took a while but I managed to create a SIGBUS crash by panning the map around.

**Comment by johnd0e** [comment]
Still unable to reproduce.
Please specify more details.
OS and browser (with versions), steps yo reproduce, etc.

**Comment by sgofferj** [comment]
- Browser (with version): Chrome Version 89.0.4389.82 (Official Build) (64-bit)
- OS/Platform (with version): Ubuntu 20.04
- Ryzen 7-1700X, 16GB RAM

I can reliably reproduce it by just grabbing the map and panning it around with the mouse. Sometimes it freezes after just a few moves, sometimes I need to move it around a bit longer. The more markers, the faster it freezes. As can be seen on the screenshot, also the icons disappear.

Can it have something to do with me using a 4K screen?

**Comment by johnd0e** [comment]
But your plnkr sandbox has not so many markers.

> Can it have something to do with me using a 4K screen?

I suppose it is important too, as well as browser window size (e.g. plnkr sandbox window is much smaller than fullscreen, does it matter?)

It also may be browser bug, so it may sense to check it with other versions (and Firefox)

**Comment by sgofferj** [comment]
I can make it freeze also in the Plunker window with the saved JSON. I just tried with firefox and I can also make it freeze there but it recovers after 2 or 3 seconds without warning popups. Also, moving the mouse pointer out of the browser window while "holding" the map might have something to do with it. I have observed a possible pattern but I haven't got enough data to be certain. I'll try to investigate further but my knowledge of browser internal workings and Javascript is quite limited.
